plugins/scaffolder-backend/src/scaffolder/stages/prepare/github.ts

```diff
@@ -30,8 +29,12 @@ export class GithubPreparer implements PreparerBase {
     this.token = params.token;
   }
 
-  async prepare(template: TemplateEntityV1alpha1): Promise<string> {
+  async prepare(
+    template: TemplateEntityV1alpha1,
+    opts: { workingDirectory: string },
+  ): Promise<string> {
     const { protocol, location } = parseLocationAnnotation(template);
+    const { workingDirectory } = opts;
```

Comment:
I wonder if it's still worth defaulting `workingDirectory` here to `os.tmp` if it's not provided in the options too and makke the `opts` optional for backwards compatibility?

Reply:
I can change that. Should the `workingDirectory` only be passed in when that is configured then? 

It feels like we are spreading out the logic then and relying on that all the preparers does the right thing but I guess we are already doing that since they can just ignore the parameter any way.

Reply:
@benjdlambert how do you want to handle this?

Reply:
Yeah only pass it in if it exists in the router, but default it here to something sane if it's not set and make this optional too so that if anyone has their own processor it won't break the types